tests/repository.py

```diff
@@ -383,75 +348,361 @@ def tearDown(self):
-        res = butlerAB.get('foo', {'bar':1})
-        self.assertEqual(res, (obj0))
-
-        # test first-found get behavior
-        repoACfg = repoCfg(root='tests/repository/repoA', mapper=MapperForTestWriting)
-        repoBCfg = repoCfg(root='tests/repository/repoB', mapper=MapperForTestWriting)
-        repoABCfg = repoCfg(root='tests/repository/repoAB', mapper=MapperForTestWriting,
-                                parentRepoCfgs=(repoACfg, repoBCfg), parentJoin='outer')
-        butlerAB = dp.Butler(dp.Butler.cfg(repoCfg=repoABCfg))
-        try:
-            res = butlerAB.get('foo', {'bar':1})
-            self.assertTrue(False, "butlerAB.get should have thrown")
-        except dp.butlerExceptions.MultipleResults as e:
-            # the butler should have found 2 results
-            self.assertEqual(len(e.locations), 2)
-
-def suite():
-    utilsTests.init()
-    suites = []
-    suites += unittest.makeSuite(TestBasics)
-    suites += unittest.makeSuite(TestWriting)
-    suites += unittest.makeSuite(TestParentMasking)
-    suites += unittest.makeSuite(TestAggregateParent)
-    suites += unittest.makeSuite(TestPeerPut)
-    return unittest.TestSuite(suites)
-
-def run(shouldExit = False):
-    utilsTests.run(suite(), shouldExit)
+        objAbc = TestObject('abc')
+        objDef = TestObject('def')
+        objGhi = TestObject('ghi')
+        objJkl = TestObject('jkl')
+
+        repoAArgs = dp.RepositoryArgs(mode='w',
+                                      root='tests/repository/repoA', 
+                                      mapper=MapperForTestWriting)
+        butler = dp.Butler(outputs=repoAArgs)
+        butler.put(objAbc, 'foo', {'bar':1})
+        butler.put(objDef, 'foo', {'bar':2})
+
+        repoBArgs = dp.RepositoryArgs(mode='w',
+                                      root='tests/repository/repoB', 
+                                      mapper=MapperForTestWriting)
+        butler = dp.Butler(outputs=repoBArgs)
+        butler.put(objGhi, 'foo', {'bar':1}) # note different object with overlapping dataId with repoA
+        butler.put(objJkl, 'foo', {'bar':3})
+
+
+        # note repo order: A, B
+        butler = dp.Butler(inputs=('tests/repository/repoA', 'tests/repository/repoB'))
+        self.assertEqual(butler.get('foo', {'bar':1}), objAbc)
+        self.assertEqual(butler.get('foo', {'bar':2}), objDef)
+        self.assertEqual(butler.get('foo', {'bar':3}), objJkl)
+
+        # note reversed repo order: B, A
+        butler = dp.Butler(inputs=('tests/repository/repoB', 'tests/repository/repoA'))
+        self.assertEqual(butler.get('foo', {'bar':1}), objGhi)
+        self.assertEqual(butler.get('foo', {'bar':2}), objDef)
+        self.assertEqual(butler.get('foo', {'bar':3}), objJkl)
+
+
+class TestTagging(unittest.TestCase):
+    """A test case for the tagging of repository classes.
+    """
+
+    def tearDown(self):
+        if os.path.exists('tests/repository'):
+            shutil.rmtree('tests/repository')
```

Comment:
In fact, that may be necessary if we switch to a test runner that can run multiple tests from the same file in parallel.